samples/ask-akka-agent/src/main/java/akka/ask/agent/application/ConversationHistoryView.java

```diff
@@ -37,28 +37,43 @@ public QueryEffect<ConversationHistory> getSessionsByUser(String userId) { // <2
     return queryResult();
   }
 
-  @Consume.FromEventSourcedEntity(SessionEntity.class)
+  @Consume.FromEventSourcedEntity(ConversationMemory.class)
   public static class ChatMessageUpdater extends TableUpdater<Session> {
 
-    public Effect<Session> onEvent(SessionEvent event) {
+    public Effect<Session> onEvent(ConversationMemory.Event event) {
       return switch (event) {
-        case SessionEvent.AiMessageAdded added -> aiMessage(added);
-        case SessionEvent.UserMessageAdded added -> userMessage(added);
+        case ConversationMemory.Event.AiMessageAdded added -> aiMessage(added);
+        case ConversationMemory.Event.UserMessageAdded added -> userMessage(added);
+        default -> effects().ignore();
       };
     }
 
-    private Effect<Session> aiMessage(SessionEvent.AiMessageAdded added) {
-      Message newMessage = new Message(added.response(), "ai", added.timeStamp().toEpochMilli());
-      var rowState = rowStateOrNew(added.userId(), added.sessionId());
+    private Effect<Session> aiMessage(ConversationMemory.Event.AiMessageAdded added) {
+      var timestamp = System.currentTimeMillis(); // FIXME add timestamp to ConversationMemory.Event
+      Message newMessage = new Message(added.message(), "ai", timestamp);
+      var rowState = rowStateOrNew(userId(), sessionId());
       return effects().updateRow(rowState.add(newMessage));
     }
 
-    private Effect<Session> userMessage(SessionEvent.UserMessageAdded added) {
-      Message newMessage = new Message(added.query(), "user", added.timeStamp().toEpochMilli());
-      var rowState = rowStateOrNew(added.userId(), added.sessionId());
+    private Effect<Session> userMessage(ConversationMemory.Event.UserMessageAdded added) {
+      var timestamp = System.currentTimeMillis(); // FIXME add timestamp to ConversationMemory.Event
+      Message newMessage = new Message(added.message(), "user", timestamp);
+      var rowState = rowStateOrNew(userId(), sessionId());
       return effects().updateRow(rowState.add(newMessage));
     }
 
+    private String userId() {
+      var agentSessionId = updateContext().eventSubject().get();
+      int i = agentSessionId.indexOf("-");
+      return agentSessionId.substring(0, i);
```

Comment:
Clunky to encode the userId in the session id.
A real solution might have to use a KeyValue entity for mapping of sessionId to userId, and lookup the userId that way here.